include/swift/ABI/MetadataValues.h

```diff
@@ -542,17 +542,17 @@ class ExistentialTypeFlags {
 enum class FunctionMetadataConvention: uint8_t {
   Swift = 0,
   Block = 1,
-  Thin = 2,
+  Thin  = 2,
   CFunctionPointer = 3,
 };
 
 /// Flags in a function type metadata record.
 template <typename int_type>
 class TargetFunctionTypeFlags {
   enum : int_type {
-    NumParametersMask = 0x00FFFFFFU,
-    ConventionMask    = 0x0F000000U,
-    ConventionShift   = 25U,
+    NumParametersMask = 0x0000FFFFU,
+    ConventionMask    = 0x00FF0000U,
+    ConventionShift   = 16U,
     ThrowsMask        = 0x10000000U,
     ParamFlagsMask    = 0x01000000U,
```

Comment:
Are you intentionally using separate bits here instead of 0x01000000 and 0x02000000?

Also, I think we should have a comment here explaining the planned approach if we run out of bits in the FunctionTypeFlags.  (We use the last bit to say that there's another bitset stored elsewhere, and we pass it to getFunctionTypeParameter at the end of the parameter-type array.)

Reply:
Yes, I just wanted to keep them separate, but I can change it to 0x01 and 0x02 to keep them close together if that's better?

Reply:
Please.  Using adjacent bits avoids creating unnecessary problems if we someday want a multi-bit field.

Reply:
Alright, no problem!